internal/engine/ingester/artifact/artifact.go

```diff
@@ -413,3 +413,13 @@ func signerIdentityFromCertificate(c *certificate.Summary) (string, error) {
 
 	return builderURL, nil
 }
+
+func getVersioner(prov provifv1.Provider) (provifv1.ArtifactProvider, error) {
```

Comment:
Why do we need this and not just `v1.As[v1.ArtifactProvider]` ?

Reply:
Because it does not implement the `Provider` interface. However, I can do type casting, which is what I'm trying right now.